app/assets/javascripts/Components/submission_file_manager.jsx

```diff
@@ -87,15 +91,19 @@ class SubmissionFileManager extends React.Component {
       .then(this.endAction);
   };
 
-  handleCreateFiles = (files, unzip) => {
+  handleCreateFiles = (files, unzip, single_file_name = "") => {
     if (
       !this.props.starterFileChanged ||
       confirm(I18n.t("assignments.starter_file.upload_confirmation"))
     ) {
       const prefix = this.state.uploadTarget || "";
       this.setState({showUploadModal: false, uploadTarget: undefined});
       let data = new FormData();
-      Array.from(files).forEach(f => data.append("new_files[]", f, f.name));
+      if (files.length === 1) {
```

Comment:
You need an extra check for the rename parameter being truthy (so it isn't `undefined`/`null`/empty string). A common idiom we use for this is `!!single_file_name`, which will convert the value into a boolean explicitly